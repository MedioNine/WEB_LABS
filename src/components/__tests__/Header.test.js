import React from 'react';
import ReactTestUtils from 'react-dom/test-utils'; 
import Header from '../HeaderComponent';
import { shallow } from 'enzyme';


describe('HeaderComponent', () => {
  
    const props = {
        isNavOpen: false,
        isLoginOpen: false,
        isRegOpen: false,
        errLog: "",
        errReg: "",
        username: "",
        password: "",
        Token: localStorage.getItem("Token"),
        reg: {
          username: "",
          password: "",
          password2: "",
          email: "",
        },
      };
    

      describe('Header container initial', () => {
        const headerContainer = (<Header {...props} />)

        it('renders properly', () => {
            expect(headerContainer).toMatchSnapshot()
          })    

          const component = shallow(<Header {...props} />)

        it('renders logo properly', () => {
            expect(component.find('NavbarBrand')).toHaveLength(1)
         })
         

         it('firstly no token', () => {
            expect(component.prop('Token')).toEqual(undefined)
         })
     })

     

})