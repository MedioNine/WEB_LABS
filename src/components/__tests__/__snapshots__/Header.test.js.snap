// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent Header container initial renders properly 1`] = `
<Header
  Token={null}
  errLog=""
  errReg=""
  isLoginOpen={false}
  isNavOpen={false}
  isRegOpen={false}
  password=""
  reg={
    Object {
      "email": "",
      "password": "",
      "password2": "",
      "username": "",
    }
  }
  username=""
/>
`;
